packages/@aws-cdk/aws-iam/lib/role.ts

```diff
@@ -473,6 +496,55 @@ export class Role extends Resource implements IRole {
 
     return errors;
   }
+
+  /**
+   * Split large inline policies into managed policies
+   *
+   * This gets around the 10k bytes limit on role policies.
+   */
+  private splitLargePolicy() {
+    if (!this.defaultPolicy || this._didSplit) {
+      return;
+    }
+    this._didSplit = true;
```

Comment:
Am I understanding correctly that this is a marker for an (attempted) split? The `true` holds whether or not the split actually happens.